codegen/projections/cbor_event_streams/lib/cbor_event_streams/client.rb

```diff
@@ -0,0 +1,139 @@
+      output
+    end
+
+    # @param [Hash | Types::StartEventStreamInput] params
+    #   Request parameters for this operation.
+    #   See {Types::StartEventStreamInput#initialize} for available parameters.
+    #   Do not set values for the event stream member +event+.
+    #   Instead use the returned output to signal input events.
+    # @param [Hash] options
+    #   Request option override of configuration. See {Config#initialize} for available options.
+    #   Some configurations cannot be overridden.
+    # @option options [EventStream::StartEventStreamHandler] :event_stream_handler
+    #   Event Stream Handler used to register handlers that will be called when events are received.
+    # @return [EventStream::StartEventStreamOutput]
+    #   An open stream that supports sending (signaling) input events to the service.
+    # @example Request syntax with placeholder values
+    #   resp = client.start_event_stream(
+    #     event: {
+    #       # One of:
+    #       simple_event: {
+    #         message: 'message'
+    #       },
+    #       nested_event: {
+    #         nested: {
+    #           values: [
+    #             'member'
+    #           ]
+    #         }
+    #       },
+    #       explicit_payload_event: {
+    #         header_a: 'headerA',
+    #       }
+    #     },
+    #     initial_structure: {
+    #       message: 'message',
+    #     }
+    #   )
+    # @example Response structure
+    #   resp.data #=> Types::StartEventStreamOutput
+    #   resp.data.event #=> Types::Events, one of [SimpleEvent, NestedEvent, ExplicitPayloadEvent]
+    #   resp.data.event.simple_event #=> Types::SimpleEvent
+    #   resp.data.event.simple_event.message #=> String
+    #   resp.data.event.nested_event #=> Types::NestedEvent
+    #   resp.data.event.nested_event.nested #=> Types::NestedStructure
+    #   resp.data.event.nested_event.nested.values #=> Array<String>
+    #   resp.data.event.nested_event.nested.values[0] #=> String
+    #   resp.data.event.explicit_payload_event #=> Types::ExplicitPayloadEvent
+    #   resp.data.event.explicit_payload_event.header_a #=> String
+    #   resp.data.event.explicit_payload_event.payload #=> Types::NestedStructure
+    #   resp.data.initial_structure #=> Types::InitialStructure
+    #   resp.data.initial_structure.message #=> String
+    #   resp.data.initial_structure.nested #=> Types::NestedStructure
+    def start_event_stream(params = {}, options = {})
+      middleware_opts = {}
+      middleware_opts[:event_stream_handler] = options.delete(:event_stream_handler)
+      raise ArgumentError, 'Missing `event_stream_handler`' unless middleware_opts[:event_stream_handler]
+      response_body = ::StringIO.new
+      config = operation_config(options)
+      input = Params::StartEventStreamInput.build(params, context: 'params')
+      stack = CborEventStreams::Middleware::StartEventStream.build(config, middleware_opts)
```

Comment:
Yeah, this is tricky.

RE 1 - what do you mean about every operation?  I don't really see handlers as being useful in non event stream operations.

RE 2 - I like the idea of it being on operation config.  Conceptually thats the most appropriate, but we also don't really have a mechanism for this.  Our service config should NOT have the ability to configure handlers (both because you can have multiple event stream operations and because you could call the same operation multiple times concurrently).  So we would need some way to define special opeation level config classes?

RE: 3 - We could add another middleware manually outside of the middleware module, but that would still be very special case? It woudl also mean we couldn't use the MiddlewareBuilder java class and would need to duplicate some of that behavior, but inside the client generator.

Reply:
Some other ideas:

option 4 - What about adding the event handler to context instead of passing it to the middleware?

option 5 - We could add the event handler to the generated input object (more similar to V3 syntax, it would get passed (and validated) in params instead of in options. Then no special case handling of it).